tabular/src/autogluon/tabular/models/image_prediction/image_predictor.py

```diff
@@ -127,9 +146,44 @@ def _fit(self,
 
     def _predict_proba(self, X, **kwargs):
         X = self.preprocess(X, **kwargs)
-        y_pred_proba = self.model.predict_proba(X, as_pandas=False)
+        # TODO: Add option to crash if null is present for faster predict_proba
+        null_indices = X['image'] == ''
+        if null_indices.sum() > 0:
+            if self.num_classes is None:
+                y_pred_proba = np.zeros(len(X))
+            else:
+                y_pred_proba = np.zeros((len(X), self.num_classes))
+            X = X.reset_index(drop=True)
+            null_indices = X['image'] == ''
+            X = X[~null_indices]
+            null_indices_rows = list(null_indices[null_indices].index)
+            non_null_indices_rows = list(null_indices[~null_indices].index)
+            y_pred_proba[null_indices_rows] = self._dummy_pred_proba
+            y_pred_proba[non_null_indices_rows] = self.model.predict_proba(X, as_pandas=False)
```

Comment:
is it necessary to create lists? I think updates can be done directly with index values using iloc

Reply:
y_pred_proba is not a DataFrame, its a numpy ndarray, it doesn't have an index or iloc.